lib/tools/system-calls.js

```diff
@@ -546,13 +546,20 @@ systemCallMethods.killAllEmulators = async function () {
  * @param {string} avdName - The name of the emulator to be killed.
  */
 systemCallMethods.killEmulator = async function (avdName) {
-  log.debug(`killing avd '${avdName}'`);
-  let device = await this.getRunningAVD(avdName);
-  if (device) {
-    await this.adbExec(['emu', 'kill']);
-    log.info(`successfully killed emulator '${avdName}'`);
+  if (util.hasValue(avdName)) {
+    log.debug(`killing avd '${avdName}'`);
+    let device = await this.getRunningAVD(avdName);
+    if (device) {
+      await this.adbExec(['emu', 'kill']);
+      log.info(`successfully killed emulator '${avdName}'`);
+    } else {
+      log.info(`no avd with name '${avdName}' running. skipping kill step.`);
+    }
   } else {
-    log.info(`no avd with name '${avdName}' running. skipping kill step.`);
+    // killing the current avd
+    log.debug(`killing avd with id '${this.curDeviceId}'`);
+    await this.adbExec(['emu', 'kill']);
```

Comment:
should we check if the emu is running first?

Reply:
Sure. It is checked in the other section, too.